sources/directconnect/direct-connect-gateway-attachment.go

```diff
@@ -0,0 +1,119 @@
+package directconnect
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/directconnect"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func directConnectGatewayAttachmentOutputMapper(_ context.Context, _ *directconnect.Client, scope string, _ *directconnect.DescribeDirectConnectGatewayAttachmentsInput, output *directconnect.DescribeDirectConnectGatewayAttachmentsOutput) ([]*sdp.Item, error) {
+	items := make([]*sdp.Item, 0)
+
+	for _, attachment := range output.DirectConnectGatewayAttachments {
+		attributes, err := sources.ToAttributesCase(attachment, "tags")
+		if err != nil {
+			return nil, err
+		}
+
+		// The uniqueAttributeValue for this is a custom field:
+		// {gatewayId} {virtualInterfaceId}
+		// i.e., "cf68415c-f4ae-48f2-87a7-3b52cexample dxvif-ffhhk74f"
+		err = attributes.Set("uniqueName", fmt.Sprintf(gatewayIDVirtualInterfaceIDFmt, *attachment.DirectConnectGatewayId, *attachment.VirtualInterfaceId))
+		if err != nil {
+			return nil, err
+		}
+
+		item := sdp.Item{
+			Type:            "directconnect-direct-connect-gateway-attachment",
+			UniqueAttribute: "uniqueName",
+			Attributes:      attributes,
+			Scope:           scope,
+		}
+
+		if attachment.DirectConnectGatewayId != nil {
+			// +overmind:link directconnect-direct-connect-gateway
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway",
+					Method: sdp.QueryMethod_GET,
+					Query:  *attachment.DirectConnectGatewayId,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// This is not clearly documented, but it seems that if the gateway is deleted, the attachment state will change to detaching
+					In: true,
+					// We can't affect the direct connect gateway
+					Out: false,
```

Comment:
I think you could probably argue this in both directions, but it's better to err on the side of `true` usually. If I ask ChatGPT it seems to think that changes in an interface could affect the gateway, and therefore we need the blast to propagate through the attachment

<details>
<summary>ChatGPT Response</summary>
In AWS, a Direct Connect Virtual Interface (VIF) and a Direct Connect Gateway (DX Gateway) are separate entities, but changes to a Direct Connect Virtual Interface can indirectly impact the traffic flow and connectivity associated with a Direct Connect Gateway. Let me explain how:

1. **Routing Configuration:** A Direct Connect Virtual Interface is associated with a BGP (Border Gateway Protocol) session that facilitates the exchange of routing information between your on-premises network and your AWS resources. When you make changes to the routing configuration of a Virtual Interface (e.g., adding or removing prefixes), it affects how traffic is routed between your on-premises network and the AWS resources associated with the Virtual Interface.

2. **Direct Connect Gateway Association:** If your Direct Connect Virtual Interface is associated with a Direct Connect Gateway (via attachment to a Virtual Private Gateway or Transit Gateway), changes in routing may impact the traffic flow to and from the resources in the associated VPCs or Transit Gateways. The Direct Connect Gateway relies on the routing information exchanged through the Virtual Interface.

3. **BGP Session Changes:** Any changes to the BGP session parameters, such as updating BGP communities or modifying the BGP session properties, can impact how routing decisions are made, which, in turn, can affect the Direct Connect Gateway's connectivity.

It's important to note that changes to a Direct Connect Virtual Interface are more likely to affect the specific routing and connectivity associated with that Virtual Interface rather than directly impacting the overall Direct Connect Gateway itself.

Always exercise caution when making changes to network configurations, and ensure that any modifications align with your overall networking requirements and design. It's recommended to carefully plan and test changes, especially in production environments, to minimize the risk of disruptions.
</details>

```suggestion
					Out: true,
```